Group tests issues

vchoi-hdfgroup opened the pull request.

vchoi-hdfgroup committed:
Merge pull request #48 from HDFGroup/feature/vfd_swmr

Feature/vfd swmr

Commit:
Address issue #1 and issue #3 of the group test failures.
See Kent's documentation "Designed to Fail Tests and Issues".
(A) Fix for issue #1: HDassert the < and = cases between old and new
entry length.  John will take care of the > case.
(B) Fix for issue #3:  set the cache copy of page_size again if different
    from f->shared->fs_page_size.

github-actions[bot] committed:
Committing clang-format changes

kyang2014 reviewed (approved):
I've tested the pull request at Mac and CentOS 7. The fix works fine. By changing the number of groups to other numbers or by changing the page buffer size or page space size to odd numbers like(page buffer size = 4703, space size = 2301), #2 and #4 issues in the design-fail document may appear. I communicated with Vailin. These messages seem expected.

Commit:
(A) Address issue #2 and issue #4 of the group test failures.
    See Kent's documentation "Designed to Fail Tests and Issues".
    (a) Fix for issue #2:
    --Print out meaningful message about max_lag when there is
    checksum error in loading an entry via H5C__load_entry().
    --H5C.c: H5C_protect()
    (b) Fix for issue #4:
    --Allocate more space when the copy of the index read from the metadata file is
      bigger than the existing size
    --H5Fvfd_swmr.c: H5F_vfd_swmr_reader_end_of_tick()
(B) When putting the old index into the delayed free list, use the old
    writer_index_offset instead of the current writer_index_offset
    --H5Fvfd_swmr.c: vfd_swmr_enlarge_shadow_index()
(C) When there is error form calling H5F_vfd_swmr_process_eot_queue() in
    VFD_SWMR_ENTER(err) and VFD_SWMR_LEAVE(err), should report
    FAIL instead of "err"
    --H5private.h: VFD_SWMR_ENTER and VFD_SWMR_LEAVE macros
(D) Add tests to verify issue #2 and issue #4 are fixed.

Commit:
Merge branch 'group_tests_issues' of https://github.com/vchoi-hdfgroup/hdf5 into group_tests_issues

github-actions[bot] committed:
Committing clang-format changes

kyang2014 reviewed (commented):
I've tested Vailin's new change. The only error encountered is the meaningful error
"Reader's API time exceeds max_lag ticks, suggest to increase the value of max_lag.".

After I increase the max_lag(sometimes the number should be very big), all the tests get passed.
